plugins/jobtype/src/azkaban/jobtype/connectors/gobblin/GobblinHadoopJob.java

```diff
@@ -0,0 +1,173 @@
+                + commonPropertiesFile.getAbsolutePath());
+          }
+
+          for (File f : presetFiles) {
+            if (GOBBLIN_PRESET_COMMON_PROPERTIES_FILE_NAME.equals(f.getName())) { //Don't load common one itself.
+              continue;
+            }
+
+            if (f.isFile()) {
+              Properties prop = new Properties();
+              try (InputStream commonIs = new BufferedInputStream(new FileInputStream(commonPropertiesFile));
+                  InputStream presetIs = new BufferedInputStream(new FileInputStream(f))) {
+                prop.load(commonIs);
+                prop.load(presetIs);
+
+                String presetName = f.getName().substring(0, f.getName().lastIndexOf('.')); //remove extension from the file name
+                gobblinPresets.put(GobblinPresets.fromName(presetName), prop);
+              } catch (IOException e) {
+                throw new RuntimeException(e);
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+
+  /**
+   * If input parameter has preset value, it will load set of properties into job property for the Gobblin job.
+   * Also, if user wants to validates the job properties(enabled by default), it will validate it based on the preset where
+   * preset is basically used as a proxy to the use case.
+   */
+  private void loadPreset() {
+    String presetName = jobProps.get(GobblinConstants.GOBBLIN_PRESET_KEY);
+    if (presetName == null) {
+      return;
+    }
+
+    GobblinPresets preset = GobblinPresets.fromName(presetName);
+    Properties presetProperties = gobblinPresets.get(preset);
+    if (presetProperties == null) {
+      throw new IllegalArgumentException("Preset " + presetName + " is not supported. Supported presets: "
+          + gobblinPresets.keySet());
+    }
+
+    getLog().info("Loading preset " + presetName + " : " + presetProperties);
+    Map<String, String> skipped = Maps.newHashMap();
+    for (String key : presetProperties.stringPropertyNames()) {
+      if (jobProps.containsKey(key)) {
+        skipped.put(key, jobProps.get(key));
+        continue;
+      }
+      jobProps.put(key, presetProperties.getProperty(key));
+    }
+    getLog().info("Loaded preset " + presetName);
+    if (!skipped.isEmpty()) {
+      getLog().info("Skipped some properties from preset as already exists in job properties. Skipped: " + skipped);
+    }
+
+    if (jobProps.getBoolean(GobblinConstants.GOBBLIN_PROPERTIES_HELPER_ENABLED_KEY, true)) {
+      getValidator(preset).validate(jobProps);
+    }
+  }
+
+  /**
+   * Transform property to make it work for Gobblin.
+   *
+   * e.g: Gobblin fails when there's semicolon in SQL query as it just appends " and 1=1;" into the query
+   * and makes the syntax incorrect. As having semicolon is a correct syntax, Azkaban will remove it for user
+   * to make it work with Gobblin.
+   */
+  private void transformProperties() {
```

Comment:
Good point.
Gobblin expects one SELECT sql statement and it fails if there's semicolon in the statement as Gobblin appends itself.

Let me keep the logic to remove semicolon -- as having semicolon is correct syntax -- but update it to validate to make sure that there's only one statement.